akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/MultiNodeClusterShardingSpec.scala

```diff
@@ -72,67 +105,112 @@ abstract class MultiNodeClusterShardingSpec(val config: MultiNodeClusterSharding
     super.afterTermination()
   }
 
-  protected def join(from: RoleName, to: RoleName): Unit = {
+  /**
+   * Flexible cluster join pattern usage.
+   *
+   * @param from the node the `Cluster.join` is `runOn`
+   * @param to the node to join to
+   * @param assert if disabled - false, the joining member's post-join status assertions
+   *               are not run. This allows tests that were not doing assertions or
+   *               doing them after `onJoinedRunOnFrom` to have more flexibility.
+   *               Defaults to true, and running member status checks.
+   * @param onJoinedRunOnFrom allows you to execute a function after join validation
+   *                          is successful, e.g. starting sharding
+   */
+  protected def join(from: RoleName, to: RoleName, onJoinedRunOnFrom: => Unit = Unit, assert: Boolean = true): Unit = {
     runOn(from) {
-      Cluster(system).join(node(to).address)
-      awaitAssert {
-        Cluster(system).state.isMemberUp(node(from).address)
+      cluster.join(node(to).address)
+      if (assert) {
+        within(20.seconds) {
+          awaitAssert {
+            // let's pick one or two vs have 3
+            // this was here originally and used in one or two tests
+            cluster.state.isMemberUp(node(from).address)
+            // these two were in a few tests
+            cluster.state.members.unsorted.map(_.uniqueAddress) should contain(cluster.selfUniqueAddress)
+            cluster.state.members.unsorted.map(_.status) shouldEqual Set(MemberStatus.Up)
+          }
+        }
       }
+      onJoinedRunOnFrom
```

Comment:
would it be good with a barrier before this `onJoinedRunOnFrom`?

I'd prefer a more explicit function `() => Unit`

Reply:
Possibly, though none of the tests were doing that.
The patterns were:
* join, barrier
* join, assert, barrier
* join, startSharding, barrier
* join, startSharding, assert, barrier

Reply:
ok, for startSharding it shouldn't matter

Reply:
`() => Unit` added